velox/functions/iceberg/tests/IcebergFunctionBaseTest.h

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "velox/functions/iceberg/Register.h"
+#include "velox/functions/prestosql/tests/utils/FunctionBaseTest.h"
+#include "velox/parse/TypeResolver.h"
+
+namespace facebook::velox::functions::iceberg::test {
+
+class IcebergFunctionBaseTest
+    : public facebook::velox::functions::test::FunctionBaseTest {
+ protected:
+  static void SetUpTestCase() {
+    parse::registerTypeResolver();
```

Comment:
Why not call FunctionBaseTest::SetUpTestCase() ?

Reply:
I see... but it is confusing and error prone to use base class like this. You need to either not use the base class or calls its Setup/Teardown functions.

Reply:
FunctionBaseTest uses SetUpTestCase, if I use setUp, I will register functions several times, because one test suite may have several test cases.

Reply:
Please, take a look.